### PR TITLE
[Relay][Refactor][std::string --> String] Relay updated with String

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
Refer #5490

@jroesch , @tqchen , @zhiics : Please help review!

### CODE REVIEW

**Comment by ANSHUMAN87** [comment]
Conflict is resolved now, Thanks!

**Comment by ANSHUMAN87** [comment]
Gentle Ping @jroesch , @tqchen , @zhiics , Thanks!

File: src/relay/backend/compile_engine.cc [hunk 1]
```diff
@@ -580,7 +580,7 @@ class CompileEngineImpl : public CompileEngineNode {
         auto symbol_name = src_func->GetAttr<String>(tvm::attr::kGlobalSymbol);
         CHECK(symbol_name.defined()) << "No external symbol is set for:\n"
                                      << AsText(src_func, false);
-        auto gv = GlobalVar(std::string(symbol_name.value()));
+        auto gv = GlobalVar(String(symbol_name.value()));
```
Comment (zhiics) [inline src/relay/backend/compile_engine.cc:583]: just `GlobalVar(symbol_name)`?

**Comment by ANSHUMAN87** [comment]
@zhiics : Thanks for review! Your comment is handled now!

**Comment by ANSHUMAN87** [comment]
Gentle ping @jroesch , @tqchen !

File: include/tvm/relay/base.h [hunk 2]
```diff
@@ -93,7 +93,7 @@ class IdNode : public Object {
    *  this only acts as a hint to the user,
    *  and is not used for equality.
    */
-  std::string name_hint;
+  String name_hint;
```
Comment (tqchen) [inline include/tvm/relay/base.h:96]: per https://github.com/apache/incubator-tvm/issues/5490#issuecomment-624408382 every update to the IRNode object needs to have a backward compatible update code

**Comment by ANSHUMAN87** [comment]
@tqchen : Your comment is handled now, added test case as well. Please check. Thanks!

**Comment by tqchen** [comment]
Thanks @ANSHUMAN87 @jroesch @zhiics

**Pull request merged**